model/dependency.go

```diff
@@ -91,6 +91,30 @@ func (di *dependencyIncluder) handle(pair TVPair, activationInfo *specificActiva
 		return true, nil
 	}
 
+	// For a task group task inside a single host task group, the previous
+	// tasks in the task group are implicit dependencies of the current task.
+	if tg := di.Project.FindTaskGroupForTask(pair.Variant, pair.TaskName); tg != nil && tg.MaxHosts == 1 {
+		catcher := grip.NewBasicCatcher()
+		for _, t := range tg.Tasks {
+			// When we reach the current task, stop recursing.
```

Comment:
Can we also mention that the task itself is already included, so there's no need to `handle` it as well? Would clarify why the task in the task group doesn't need to be handled.